src/common/util.cpp

```diff
@@ -79,3 +76,36 @@ std::vector<uint8_t> hexToBytes(std::string hexStr) {
 
     return bytes;
 }
+
+const int compareFloats(const float a, const float b, const size_t epsilon) {
+    /*
+    A simple decimal comparison might involve subtraction, fabs, and the result being less than a smaller decimal.
+    This method converts the IEEE float values into lexicographically-ordered two's-complement integers.
+    If (float)a > (float)b then (int)a > (int)b; but, precision concerns are lessened.
+    The error becomes a count of "the maximum number of lexico-numbers allowed between equal lexico-a and lexico-b."
+    To avoid encountering a NaN value, error can not risk being too big, however.
+    */
+    if(epsilon >= 4194304) {
```

Comment:
This is actually borrowed from a collaborated project I haven't touched since my college days.  I do recall someone else used the basic convert and dereference method and here I just reframed.  I recall that float NaNs were the reason but the specifics beyond "avoiding a float NaN value" are lost on me right now.

Before there's any kind of sign-off on this, let me see what I can dig up by searching for keywords that get jogged from my memory.